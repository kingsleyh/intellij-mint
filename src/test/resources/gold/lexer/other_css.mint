component Css {

  style rangeSlider {
    width: 100%;
    margin: 60px 0 0 0%;
  }

  style sliderRange {
    width: 100%;
    margin: 60px 0 0 0%;
    -webkit-appearance: none;
    width: calc(100% - (73px));
    height: 10px;
    border-radius: 5px;
    background: #d7dcdf;
    outline: none;
    padding: 0;
    margin: 0;

    &::-webkit-slider-thumb {
      -webkit-appearance: none;
      appearance: none;
      width: 20px;
      height: 20px;
      border-radius: 50%;
      background: #2c3e50;
      cursor: pointer;
      transition: background .15s ease-in-out;
    }

    &::-webkit-slider-thumb:hover {
      background: #1abc9c;
    }

    &::-webkit-slider-thumb {
      background: #1abc9c;
    }

    &::-moz-range-thumb {
      width: 20px;
      height: 20px;
      border: 0;
      border-radius: 50%;
      background: #2c3e50;
      cursor: pointer;
      transition: background .15s ease-in-out;
    }

    &::-moz-range-thumb:hover {
      background: #1abc9c;
    }

    &:active::-moz-range-thumb {
      background: #1abc9c;
    }

    &:focus::-webkit-slider-thumb {
      box-shadow: 0 0 0 3px #fff, 0 0 0 6px #1abc9c;
    }

    &::-moz-range-track {
      background: #d7dcdf;
      border: 0;
    }

    &:input::-moz-focus-inner,
    &:input::-moz-focus-outer {
      border: 0;
    }
  }

  style rangeValue {
    display: inline-block;
    position: relative;
    width: 60px;
    color: #fff;
    line-height: 20px;
    text-align: center;
    border-radius: 3px;
    background: #2c3e50;
    padding: 5px 10px;
    margin-left: 8px;

    &:after {
      position: absolute;
      top: 8px;
      left: -7px;
      width: 0;
      height: 0;
      border-top: 7px solid transparent;
      border-right: 7px solid #2c3e50;
      border-bottom: 7px solid transparent;
      content: '';
    }
  }


  fun render : Html {
    </div>
  }

}