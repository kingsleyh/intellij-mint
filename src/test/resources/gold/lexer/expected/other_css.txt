component ('component')
WHITE_SPACE (' ')
IDENT ('Css')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n\n  ')
IDENT ('style')
WHITE_SPACE (' ')
IDENT ('rangeSlider')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n    ')
IDENT ('width')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('100')
% ('%')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('margin')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('60')
IDENT ('px')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
% ('%')
; (';')
WHITE_SPACE ('\n  ')
} ('}')
WHITE_SPACE ('\n\n  ')
IDENT ('style')
WHITE_SPACE (' ')
IDENT ('sliderRange')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n    ')
IDENT ('width')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('100')
% ('%')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('margin')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('60')
IDENT ('px')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
% ('%')
; (';')
WHITE_SPACE ('\n    ')
- ('-')
IDENT ('webkit')
- ('-')
IDENT ('appearance')
: (':')
WHITE_SPACE (' ')
IDENT ('none')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('width')
: (':')
WHITE_SPACE (' ')
IDENT ('calc')
( ('(')
NUMBER_LITERAL ('100')
% ('%')
WHITE_SPACE (' ')
- ('-')
WHITE_SPACE (' ')
( ('(')
NUMBER_LITERAL ('73')
IDENT ('px')
) (')')
) (')')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('height')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('10')
IDENT ('px')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('border')
- ('-')
IDENT ('radius')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('5')
IDENT ('px')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('background')
: (':')
WHITE_SPACE (' ')
# ('#')
IDENT ('d7dcdf')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('outline')
: (':')
WHITE_SPACE (' ')
IDENT ('none')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('padding')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('margin')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
; (';')
WHITE_SPACE ('\n\n    ')
& ('&')
:: ('::')
- ('-')
IDENT ('webkit')
- ('-')
IDENT ('slider')
- ('-')
IDENT ('thumb')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
- ('-')
IDENT ('webkit')
- ('-')
IDENT ('appearance')
: (':')
WHITE_SPACE (' ')
IDENT ('none')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('appearance')
: (':')
WHITE_SPACE (' ')
IDENT ('none')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('width')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('20')
IDENT ('px')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('height')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('20')
IDENT ('px')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('border')
- ('-')
IDENT ('radius')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('50')
% ('%')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('background')
: (':')
WHITE_SPACE (' ')
# ('#')
NUMBER_LITERAL ('2')
IDENT ('c3e50')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('cursor')
: (':')
WHITE_SPACE (' ')
IDENT ('pointer')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('transition')
: (':')
WHITE_SPACE (' ')
IDENT ('background')
WHITE_SPACE (' ')
. ('.')
NUMBER_LITERAL ('15')
IDENT ('s')
WHITE_SPACE (' ')
IDENT ('ease')
- ('-')
IDENT ('in')
- ('-')
IDENT ('out')
; (';')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n    ')
& ('&')
:: ('::')
- ('-')
IDENT ('webkit')
- ('-')
IDENT ('slider')
- ('-')
IDENT ('thumb')
: (':')
IDENT ('hover')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
IDENT ('background')
: (':')
WHITE_SPACE (' ')
# ('#')
NUMBER_LITERAL ('1')
IDENT ('abc9c')
; (';')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n    ')
& ('&')
:: ('::')
- ('-')
IDENT ('webkit')
- ('-')
IDENT ('slider')
- ('-')
IDENT ('thumb')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
IDENT ('background')
: (':')
WHITE_SPACE (' ')
# ('#')
NUMBER_LITERAL ('1')
IDENT ('abc9c')
; (';')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n    ')
& ('&')
:: ('::')
- ('-')
IDENT ('moz')
- ('-')
IDENT ('range')
- ('-')
IDENT ('thumb')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
IDENT ('width')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('20')
IDENT ('px')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('height')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('20')
IDENT ('px')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('border')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('border')
- ('-')
IDENT ('radius')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('50')
% ('%')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('background')
: (':')
WHITE_SPACE (' ')
# ('#')
NUMBER_LITERAL ('2')
IDENT ('c3e50')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('cursor')
: (':')
WHITE_SPACE (' ')
IDENT ('pointer')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('transition')
: (':')
WHITE_SPACE (' ')
IDENT ('background')
WHITE_SPACE (' ')
. ('.')
NUMBER_LITERAL ('15')
IDENT ('s')
WHITE_SPACE (' ')
IDENT ('ease')
- ('-')
IDENT ('in')
- ('-')
IDENT ('out')
; (';')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n    ')
& ('&')
:: ('::')
- ('-')
IDENT ('moz')
- ('-')
IDENT ('range')
- ('-')
IDENT ('thumb')
: (':')
IDENT ('hover')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
IDENT ('background')
: (':')
WHITE_SPACE (' ')
# ('#')
NUMBER_LITERAL ('1')
IDENT ('abc9c')
; (';')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n    ')
& ('&')
: (':')
IDENT ('active')
:: ('::')
- ('-')
IDENT ('moz')
- ('-')
IDENT ('range')
- ('-')
IDENT ('thumb')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
IDENT ('background')
: (':')
WHITE_SPACE (' ')
# ('#')
NUMBER_LITERAL ('1')
IDENT ('abc9c')
; (';')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n    ')
& ('&')
: (':')
IDENT ('focus')
:: ('::')
- ('-')
IDENT ('webkit')
- ('-')
IDENT ('slider')
- ('-')
IDENT ('thumb')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
IDENT ('box')
- ('-')
IDENT ('shadow')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
WHITE_SPACE (' ')
NUMBER_LITERAL ('3')
IDENT ('px')
WHITE_SPACE (' ')
# ('#')
IDENT ('fff')
, (',')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
WHITE_SPACE (' ')
NUMBER_LITERAL ('6')
IDENT ('px')
WHITE_SPACE (' ')
# ('#')
NUMBER_LITERAL ('1')
IDENT ('abc9c')
; (';')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n    ')
& ('&')
:: ('::')
- ('-')
IDENT ('moz')
- ('-')
IDENT ('range')
- ('-')
IDENT ('track')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
IDENT ('background')
: (':')
WHITE_SPACE (' ')
# ('#')
IDENT ('d7dcdf')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('border')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
; (';')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n\n    ')
& ('&')
: (':')
IDENT ('input')
:: ('::')
- ('-')
IDENT ('moz')
- ('-')
IDENT ('focus')
- ('-')
IDENT ('inner')
, (',')
WHITE_SPACE ('\n    ')
& ('&')
: (':')
IDENT ('input')
:: ('::')
- ('-')
IDENT ('moz')
- ('-')
IDENT ('focus')
- ('-')
IDENT ('outer')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
IDENT ('border')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
; (';')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n  ')
} ('}')
WHITE_SPACE ('\n\n  ')
IDENT ('style')
WHITE_SPACE (' ')
IDENT ('rangeValue')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n    ')
IDENT ('display')
: (':')
WHITE_SPACE (' ')
IDENT ('inline')
- ('-')
IDENT ('block')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('position')
: (':')
WHITE_SPACE (' ')
IDENT ('relative')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('width')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('60')
IDENT ('px')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('color')
: (':')
WHITE_SPACE (' ')
# ('#')
IDENT ('fff')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('line')
- ('-')
IDENT ('height')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('20')
IDENT ('px')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('text')
- ('-')
IDENT ('align')
: (':')
WHITE_SPACE (' ')
IDENT ('center')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('border')
- ('-')
IDENT ('radius')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('3')
IDENT ('px')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('background')
: (':')
WHITE_SPACE (' ')
# ('#')
NUMBER_LITERAL ('2')
IDENT ('c3e50')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('padding')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('5')
IDENT ('px')
WHITE_SPACE (' ')
NUMBER_LITERAL ('10')
IDENT ('px')
; (';')
WHITE_SPACE ('\n    ')
IDENT ('margin')
- ('-')
IDENT ('left')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('8')
IDENT ('px')
; (';')
WHITE_SPACE ('\n\n    ')
& ('&')
: (':')
IDENT ('after')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n      ')
IDENT ('position')
: (':')
WHITE_SPACE (' ')
IDENT ('absolute')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('top')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('8')
IDENT ('px')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('left')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('-7')
IDENT ('px')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('width')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('height')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('0')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('border')
- ('-')
IDENT ('top')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('7')
IDENT ('px')
WHITE_SPACE (' ')
IDENT ('solid')
WHITE_SPACE (' ')
IDENT ('transparent')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('border')
- ('-')
IDENT ('right')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('7')
IDENT ('px')
WHITE_SPACE (' ')
IDENT ('solid')
WHITE_SPACE (' ')
# ('#')
NUMBER_LITERAL ('2')
IDENT ('c3e50')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('border')
- ('-')
IDENT ('bottom')
: (':')
WHITE_SPACE (' ')
NUMBER_LITERAL ('7')
IDENT ('px')
WHITE_SPACE (' ')
IDENT ('solid')
WHITE_SPACE (' ')
IDENT ('transparent')
; (';')
WHITE_SPACE ('\n      ')
IDENT ('content')
: (':')
WHITE_SPACE (' ')
BAD_CHARACTER (''')
BAD_CHARACTER (''')
; (';')
WHITE_SPACE ('\n    ')
} ('}')
WHITE_SPACE ('\n  ')
} ('}')
WHITE_SPACE ('\n\n\n  ')
fun ('fun')
WHITE_SPACE (' ')
IDENT ('render')
WHITE_SPACE (' ')
: (':')
WHITE_SPACE (' ')
IDENT ('Html')
WHITE_SPACE (' ')
{ ('{')
WHITE_SPACE ('\n    ')
</ ('</')
IDENT ('div')
> ('>')
WHITE_SPACE ('\n  ')
} ('}')
WHITE_SPACE ('\n\n')
} ('}')